### ISSUE TITLE
Deprecations & upcoming breaking changes

### ISSUE THREAD

**Issue by owensgit** [Legal-and-General/canopy#315]
ℹ️  &nbsp; Small breaking changes are more easily consumed when shipped together in a major version bump.

💡  &nbsp; When identifying such a change, such as removing a CSS var, add a deprecation notice and where possible provide deprecation warnings in the console.

💬   &nbsp; Add a comment to this issue to log current deprecations and upcoming minor breaking changes.

🚀  &nbsp; The pull request should be merged into the `next` branch, where it can be part of the next **pre-release**. At an appropriate time, `next` can be merged into `master` creating a major release of Canopy (with some nice release notes).

♻️  &nbsp; When released, this issue can be closed down and another one spun up for the next batch.

**Comment by owensgit** [comment]
**Deprecated SCSS import in the following files**

- projects/canopy/src/lib/alert/alert.component.scss
- projects/canopy/src/lib/details/details.component.scss
- projects/canopy/src/lib/forms/validation/validation.component.scss

These files currently import `variants.scss` to make the global variant classes available in these components.

These imports need to be removed and a breaking change needs to be flagged so projects know that they need to import the new variants.scss file to get the variant classes, such as `.lg-variant--info`.

**Comment by andy-polhill** [comment]
(temporarily removing the pin in favour of #332, max of 3 pins, will swap it back once the other `next` branch is merged)

**Comment by andy-polhill** [comment]
I believe the next schedules breaking change will be the Angular 12 upgrade.

**Comment by owensgit** [comment]
**Breaking change:** Removed solid-secondary button variant: https://github.com/Legal-and-General/canopy/pull/451

**Comment by elenagarrone** [comment]
Deprecated: `LgSortCodeComponent`. I'll create a PR soon to remove the component in the next breaking change.

**Comment by owensgit** [comment]
**Breaking change:** removal of the pipes module: https://github.com/Legal-and-General/canopy/pull/610/commits
